lib/server/server.js

```diff
@@ -60,19 +60,14 @@ Meteor.methods({
         updatedAt: new Date()
       };
 
-      if (options.id) {
-        // XXX: We might want to check the id - Why isnt there a match for id
-        // in the Meteor check... Normal length 17 (could be larger), and
-        // numbers+letters are used in Random.id() with exception of 0 and 1
-        doc._id = options.id;
-        // The user wanted us to use a specific id, we didn't find this while
-        // searching. The client could depend on the id eg. as reference so
-        // we respect this and try to create a document with the selected id;
-        Push.appCollection._collection.insert(doc);
-      } else {
-        // Get the id from insert
-        doc._id = Push.appCollection.insert(doc);
-      }
+      // XXX: We might want to check the id - Why isnt there a match for id
+      // in the Meteor check... Normal length 17 (could be larger), and
+      // numbers+letters are used in Random.id() with exception of 0 and 1
+      doc._id = options.id || Random.id();
+      // The user wanted us to use a specific id, we didn't find this while
+      // searching. The client could depend on the id eg. as reference so
+      // we respect this and try to create a document with the selected id;
+      Push.appCollection._collection.insert(doc);
```

Comment:
Ok, cool. thanks!

Adam Ginsburg
adam.ginsburg@buzzy.buzz
tel: +61 411 151 732
www.buzzy.buzz



> On 11 May 2016, at 6:02 PM, Morten N.O. Nørgaard Henriksen <notifications@github.com> wrote:
> 
> In lib/server/server.js <https://github.com/raix/push/pull/203#discussion_r62806039>:
> 
> > -        // The user wanted us to use a specific id, we didn't find this while
> > -        // searching. The client could depend on the id eg. as reference so
> > -        // we respect this and try to create a document with the selected id;
> > -        Push.appCollection._collection.insert(doc);
> > -      } else {
> > -        // Get the id from insert
> > -        doc._id = Push.appCollection.insert(doc);
> > -      }
> > +      // XXX: We might want to check the id - Why isnt there a match for id
> > +      // in the Meteor check... Normal length 17 (could be larger), and
> > +      // numbers+letters are used in Random.id() with exception of 0 and 1
> > +      doc._id = options.id || Random.id();
> > +      // The user wanted us to use a specific id, we didn't find this while
> > +      // searching. The client could depend on the id eg. as reference so
> > +      // we respect this and try to create a document with the selected id;
> > +      Push.appCollection._collection.insert(doc);
> @adamgins <https://github.com/adamgins> I've removed the part depending on the meteor api to return id
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/raix/push/pull/203/files/9e370298edacc689f039fefa20cfd2f681631b85#r62806039>